Magento_Theme/web/js/view/custom-messages.js

```diff
@@ -0,0 +1,52 @@
+define([
+  'jquery',
+  'uiComponent',
+  'Magento_Customer/js/customer-data',
+  'underscore',
+  'escaper',
+  'jquery/jquery-storageapi',
+], function ($, Component, customerData, _, escaper) {
+  'use strict';
+
+  return Component.extend({
+    defaults: {
+      cookieMessages: [],
+      messages: [],
+      allowedTags: ['div', 'span', 'b', 'strong', 'i', 'em', 'u', 'a'],
+    },
+
+    initialize() {
+      this._super();
+
+      this.cookieMessages = _.unique(
+        $.cookieStorage.get('mage-messages'),
+        'text'
+      );
+
+      this.messages = customerData.get('messages').extend({
+        disposableCustomerData: 'messages',
+      });
+
+      if (!_.isEmpty(this.messages().messages)) {
+        customerData.set('messages', {});
+      }
+
+      $.mage.cookies.set('mage-messages', '', {
+        samesite: 'strict',
+        domain: '',
+      });
+
+      this.closeMessage();
+    },
+
+    prepareMessageForHtml(message) {
+      return escaper.escapeHtml(message, this.allowedTags);
+    },
+
+    closeMessage() {
+      $('body').on('click', '.close-message-icon', function () {
```

Comment:
don't use jQuery for interactions with Knockout's templates. find a way how to close them by knockout bindings like
```html
<button data-bind="click: closeMessage" />
```
in this case I think you have to duplicate the existing template with some changes